lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -2210,9 +2219,14 @@ selectCoinsForQuit ctx@ApiLayer{..} (ApiT walletId) = do
         action <- WD.quitStakePoolDelegationAction
             db currentEpochSlotting withdrawal
         let changeAddrGen = W.defaultChangeAddressGen (delegationAddressS @n)
+
+        optionalVoteAction <-
+            liftIO $ W.handleVotingWhenMissingInConway era db True
+
         let txCtx = defaultTransactionCtx
                 { txDelegationAction = Just action
                 , txWithdrawal = withdrawal
+                , txVotingAction = optionalVoteAction
```

Comment:
```suggestion

        let txCtx = defaultTransactionCtx
                { txDelegationAction = Just action
                , txWithdrawal = withdrawal
```

We do not need to supply a `txVotingAction ≠ Nothing` here, because the result of `WD.quitStakePoolDelegationAction` can only be `Quit`. There is no point in adding a vote to `Abstain` if we want to quit anyway. 🤔

We do need the `txDeposit`, however — the code that constructs the certificate corresponding to the `Quit` action will refuse to do so without a deposit size. 😲

Reply:
indeed, txDeposit is needed when registration and derigistration happens. I do think when quitPools happens and there is voting then we need to emit user error ....

Reply:
> I do think when quitPools happens and there is voting then we need to emit user error ....

As the "Quit" endpoint goes, I think it's better to not emit an error when the wallet is voting, because this is a legacy endpoint and doesn't expect anything about voting anyway.

As far as this function goes, I think this would not happen anyway? 🤔 

* `handleVotingWhenMissingInConway` doesn't emit an error.
* The only reason why there may be an error is when `txDeposit = Nothing` and `txDelegationAction = Just Quit` and the function `ertificateFromDelegationAction ` will hit the following case:

```hs
       (Quit, Nothing) ->
           error "certificateFromDelegationAction: deposit value required in \
                 \Conway era when registration is carried out (quitting)"
```

I would like to not hit that case, because that's a bug — we didn't provide the deposit value.

Reply:
> txDeposit is needed when registration and derigistration happens

Related: https://cardanofoundation.atlassian.net/browse/ADP-3270

To properly handle:

```haskell
-- | De-Register the staking credential. Deposit, if present, must match the amount
-- that was left as a deposit upon stake credential registration.
```

https://github.com/IntersectMBO/cardano-ledger/blob/53a64c898408966940c66fa4b29d42f2656240f4/eras/conway/impl/src/Cardano/Ledger/Conway/TxCert.hs#L391-L392

i.e., the deposit not necessarily being `currentPP ^. ppKeyDepositL`, I believe we'd need to either query it from the node if possible or track it in our DB.

Reply:
> track it in our DB.

😵‍💫 I think that's what we have to do — but perhaps not as part of the Conway hardfork.

The ledger still recognizes the old way of unregistering certificates, it's just that `Cardano.Api` does not expose it as part of `makeStakeAddressUnregistrationCertificate`. It's possible to use the constructor `ConwayCertificate` directly, though.